File: 01x10/c-002/main.js
function fibEvenProc(max) {
    let sum = 0, prev = 0, next = 1
    while (next < max) {
        const current = prev + next
        if (current % 2 == 0) {
            sum += current
        }
        prev = next
        next = current
    }
    return sum
}

console.log(fibEvenProc(4_000_000))

File: 01x10/c-002/main.go
package main

import "fmt"

func FibEvenProc(max int) int {
	sum, prev, next := 0, 0, 1
	for next < max {
		current := (prev + next)
		next, prev = current, next
		if current%2 == 0 {
			sum += current
		}
	}
	return sum
}

func main() {
	fmt.Println(FibEvenProc(4_000_000))
}
